pkg/trustedresources/verify.go

```diff
@@ -40,11 +43,28 @@ const (
 
 // VerifyTask verifies the signature and public key against task.
 // source is from ConfigSource.URI, which will be used to match policy patterns. k8s is used to fetch secret from cluster
-func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface, source string, policies []*v1alpha1.VerificationPolicy) error {
-	matchedPolicies, err := matchedPolicies(taskObj.TaskMetadata().Name, source, policies)
+func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface, source string, verificationpolicies []*v1alpha1.VerificationPolicy) error {
+	matchedPolicies, err := GetMatchedPolicies(taskObj.TaskMetadata().Name, source, verificationpolicies)
 	if err != nil {
-		return err
+		if errors.Is(err, ErrNoMatchedPolicies) {
+			// skip the verification if no policies are found and trusted-resources-verification-no-match-policy is set to ignore or warn
+			switch config.GetVerificationNoMatchPolicy(ctx) {
+			case config.IgnoreNoMatchPolicy:
+				return nil
+			case config.WarnNoMatchPolicy:
+				logger := logging.FromContext(ctx)
+				logger.Warnf("failed to get matched policies: %v", err)
+				return nil
+			case config.FailNoMatchPolicy:
+				return fmt.Errorf("VerifyTask failed to get matched policies: %w", err)
```

Comment:
will this get returned back to the user? if so remove the VerifyTask bit. You could probably just omit this case and let it fallthrough the ErrNoMatchedPolicies check.

Reply:
changed it to fallthrough, but not sure "remove the VerifyTask bit" means?